plugins/Lights/plugin.cpp

```diff
@@ -18,14 +18,33 @@
 
 #include "plugin.h"
 #include "Lights.h"
+#include "HfdLights.h"
+#include "LegacyLights.h"
+
+// libandroid-properties
+#include <hybris/properties/properties.h>
 
 #include <QtQml/qqml.h>
 
 static QObject *lights_provider(QQmlEngine *engine, QJSEngine *scriptEngine)
 {
     Q_UNUSED(engine)
     Q_UNUSED(scriptEngine)
-    return new Lights();
+
+    bool use_hfd = false;
+    char buffer[20];
```

Comment:
Use `PROP_VALUE_MAX` for the array size. In practice it probably won't exceed, but to be safe.